### PR TITLE
Prefer `ctest` and `cmake --install` over generated targets

### PR DESCRIPTION
**Pull request by dominiklohmann**
### :notebook_with_decorative_cover: Description

This PR contains a bunch of things left out of #1657 intentionally to keep that PR simple.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit, or file-by-file when enabling "hide whitespace changes".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ Once you have all dependencies in place, build VAST with the following commands:
 ```sh
 cmake -B build
 cmake --build build
-cmake --build build --target test
```
Comment (lava) [inline README.md:89]: Not sure how you can live with the broken symmetry for this one :D

**Review by lava — APPROVED** [review]
These all make sense to me.

File: README.md (hunk 1 above)
Comment (dominiklohmann) [reply README.md:89]: I'm not sure how I'll manage, might need to get some sort of emotional support. 🤕

**Comment by lava** [comment]
Actually, addendum after looking at the CI results: The gcc warning is not a false positive imho, this is a real bug in `data_processor.hpp`. We just can't do much about it since it's in caf. So at least the commit message/comment should be reworded.

**Pull request merged**